src/quicklist.c

```diff
@@ -467,6 +468,17 @@ _quicklistNodeSizeMeetsOptimizationRequirement(const size_t sz,
 
 #define sizeMeetsSafetyLimit(sz) ((sz) <= SIZE_SAFETY_LIMIT)
 
+/* Calculate the size limit or length limit of the quicklist node
+ * based on 'fill', and is also used to limit list listpack. */
+void quicklistSizeAndCountLimit(int fill, size_t *size, unsigned long *count) {
+    *size = SIZE_MAX;
+    *count = ULONG_MAX;
+    if (fill >= 0)
+        *count = fill;
+    else
+        *size = optimization_level[-fill - 1];
+}
```

Comment:
Yes, I really don't like this implementation (it pollutes the quicklist).
The reason why I don't introduct `list-max-listpack-entries` is that I'm worried that if the user configures `list-max-listpack-entries` and `list-max-listpack-size` incorrectly, it might cause frequent converting between quicklist and listpack.

Like following example
```sh
CONFIG SET list-max-listpack-entries 1
CONFIG SET list-max-listpack-size 10
LPUSH lst x
1) LPUSH lst y
2) LPOP lst
repeat 1 and 2
```